.github/scripts/crib/lib/check-route53-records.js

```diff
@@ -0,0 +1,83 @@
+import {
+  Route53Client,
+  ListResourceRecordSetsCommand,
+} from "@aws-sdk/client-route-53";
+
+// us-east-1 is the global region used by Route 53.
+const route53Client = new Route53Client({ region: "us-east-1" });
+
+// Function to wait for a specified amount of time (ms).
+const wait = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
+
+/**
+ * Check if Route 53 records exist for a given Route 53 zone.
+ *
+ * @param {string} hostedZoneId The ID of the hosted zone.
+ * @param {string[]} recordNames An array of record names to check.
+ * @param {number} maxRetries The maximum number of retries.
+ * @param {number} initialBackoff The initial backoff time in milliseconds.
+ * @returns {Promise<boolean>} True if records exist, false otherwise.
+ */
+async function route53RecordsExist(
+  hostedZoneId,
+  recordNames,
+  maxRetries = 8,
+  initialBackoff = 2000 // 2 seconds
+) {
+  let attempts = 0;
+
+  console.info("Checking DNS records in Route 53...");
+  while (attempts < maxRetries) {
+    try {
+      let isTruncated = true;
+      let nextRecordName;
+      let nextRecordType;
+      let allRecordSets = [];
+
+      while (isTruncated) {
+        const params = {
+          HostedZoneId: hostedZoneId,
+          MaxItems: "300",
+          ...(nextRecordName && { StartRecordName: nextRecordName }),
+          ...(nextRecordType && { StartRecordType: nextRecordType }),
```

Comment:
It's not [clear to me](https://github.com/aws/aws-sdk-js-v3/blob/e2fb9d50d7d8e096d8cd02527141c7c86cc7321c/clients/client-route-53/src/models/models_0.ts#L7218) that it would be safe to include these properties if the value is undefined. Might make sense to just leave them as conditional.